### ISSUE TITLE
Some stations have no audio

### ISSUE THREAD

**Issue by cookieisland** [rocky4546/tvheadend-locast#7]
I am getting a problem with some Locast stations not having any audio.  I tried a channel (kpix/dabl)  got no sound. I switched to another channel (kntv/NBC) and everything was fine.  I tested this using a web browser on a Windows device with Chrome going directly to the Locast website and it worked fine. I also tested it using the Locast app on Kodi from Lunatixz and it worked there as well.

Here is my log. I have a suspicion that the stream is a bit different and shows up with "Found 0 playlists" and "no variant streams found" and thus may be a slightly different m3u8 which triggers my problem...  I tested with and without your latest commits.
```
2021-04-12 17:42:09,212-INFO: User is a free account.
2021-04-12 17:42:09,213-INFO: Starting Stations thread...
2021-04-12 17:42:09,215-INFO: FCC facilities database cache missing. Downloading the latest FCC facilities database...
Getting list of stations based on DMA...
2021-04-12 17:42:22,855-INFO: Found 50 stations for DMA 807
2021-04-12 17:42:23,306-INFO: Starting admin website on 10.8.8.7:6077
2021-04-12 17:42:23,417-INFO: Starting streaming tuner website on 10.8.8.7:5004
2021-04-12 17:42:23,439-INFO: Available tuners=4
2021-04-12 17:42:23,526-INFO: Starting EPG process...
2021-04-12 17:42:23,733-INFO: TVHeadend_Locast is now online.
Getting station info for 1612566603941...
Determining best video stream for 1612566603941...
Found 0 Playlists
No variant streams found for this station.  Assuming single stream only.
2021-04-12 17:47:38,619-INFO: Serving https://sfo.locastnet.org/proxy/sfo/kpixdabl161827372.ts (10.0s)
2021-04-12 17:47:47,964-INFO: Serving https://sfo.locastnet.org/proxy/sfo/kpixdabl161827373.ts (10.0s)
2021-04-12 17:47:57,293-INFO: Serving https://sfo.locastnet.org/proxy/sfo/kpixdabl161827374.ts (10.0s)
Getting station info for 1316...
Determining best video stream for 1316...
Found 3 Playlists
1316 will use 1280x720 resolution at 2700000bps
2021-04-12 17:48:03,262-INFO: Serving https://sfo.locastnet.org/proxy/sfo/kntv_src269712488.ts (6.0s)
2021-04-12 17:48:06,655-INFO: Serving https://sfo.locastnet.org/proxy/sfo/kpixdabl161827375.ts (10.0s)
2021-04-12 17:48:06,656-INFO: 2. Connection dropped by end device [Errno 104] Connection reset by peer
2021-04-12 17:48:06,657-INFO: 2 Locast Connection Closed
2021-04-12 17:48:07,105-INFO: Serving https://sfo.locastnet.org/proxy/sfo/kntv_src269712489.ts (6.0s)
2021-04-12 17:48:12,811-INFO: Serving https://sfo.locastnet.org/proxy/sfo/kntv_src269712490.ts (6.0s)
2021-04-12 17:48:18,609-INFO: Serving https://sfo.locastnet.org/proxy/sfo/kntv_src269712491.ts (6.0s)
2021-04-12 17:48:24,322-INFO: Serving https://sfo.locastnet.org/proxy/sfo/kntv_src269712492.ts (6.0s)
```

**Comment by rocky4546** [comment]
An interesting issue.  I did find a number of stations that had no audio.  Some times the stream would play audio, but hang the video; while other times it would play video and not have audio.  Checked the audio codec and it was the same as other good channels.  What I found was it was Locast causing the issue.  I ran the m3u8 stream the log displayed directly into VLC from Locast and the same thing happened.  So, the issue is with the stream from Locast.

I switched the app to ffmpegproxy to see if ffmpeg could cleanup the stream.  It claimed the video portion of the stream was erratic.  This was on multiple channels.  In all bad cases, it started with the statement, no video packets received in the first file.  With ffmpeg, it would start with sound only and periodically update the video frame to a still picture while continuously playing audio.

Since ffmpeg and VLC are having issues with the stream from Locast on these channels, it does not seem something that can be resolved on this end.

**Comment by rocky4546** [comment]
DABL works fine here.  It is tied to CBS and has a strong signal, so both video and audio work fine.

**Comment by rocky4546** [comment]
As a final check, I went on Locast.org and tried playing those streams from their website.  Every channel having issues also did not play at all from Locast.org.  After hitting play, the site was just spinning and never had audio or video.

**Comment by cookieisland** [comment]
I also have seen some instances where there was no video. Today when I
delved a bit deeper on the no audio channel and  I found it would work via
chrome and Kodi/lunatixz so that made me think that the Locast stream was
somewhat ok. I will investigate more in the upcoming days...

**Comment by cookieisland** [comment]
Very interesting! I will try again.

Did you try Kodi/lunatixz?   But wait.... that may not work for you...  I
had to modify Lunatixz's kodi app  to make it work... I am using his beta
repo but it has a problem for me. It uses  inputstream.adaptive which does
not work for me.  Once I removed inputstream.adaptive his app worked and I
am able to switch to using his app to view any channels that
tvheadend-locast does not play.

**Comment by rocky4546** [comment]
Checked out the Locast for Kodi by lunatixz.  BTW, I was the one that added its ability to support free accounts and made some of the changes to upgrade it to Matrix.  inputstream.adaptive worked fine for me. I played it in a Windows environment for the test, although I also tried the Ubuntu version and had similar results. Both were on Kodi Matrix. The results are... every channel that had issues on the app also had issues with kodi/locast.  Either the video froze and played audio or the video played with no audio.  I checked all channels and it was consistent across all channels. Also checked locast.org again and had the same results as before.  Suspect they made a change that is causing the issue.  Although we might be able to fix the ATSC packets that are in error, the correct answer would be to wait for locast to fix the bug on their end.

**Comment by rocky4546** [comment]
Uninstalled inputstream.adaptive and ran a test on the windows kodi.  When the locast ad played, then the video, the video did play with audio, but when I stopped the stream and restarted without the ad playing, it again had the issues.  Again, it looks like the stream has ATSC issues.

**Comment by cookieisland** [comment]
I will test some more.  BTW: in the Lunatixz Locast code I think it will automatically reinstall inputstream.adaptive via inputstreamhelper.  Therefore I modified the code so the inputstream.adaptive calls are never made by adding a 'False' condition to the if statement.  That made the Locast Kodi app work consistently for me.  But I haven't used it much because I much prefer tvheadend-locast (your features for free users and being able to pause and record Locast shows via tvheadend is important to me)

    def playLive(self, id, opt='live'):
        log('playLive, id = %s, opt = %s'%(id,opt))             
        if id == 'NEXT_SHOW': 
            found = False
            liz   = xbmcgui.ListItem(LANGUAGE(30029))
            notificationDialog(LANGUAGE(30029), time=4000)
        else:            
            found = True
            liz   = self.resolveURL(id,opt)
            #cookieisland turn off inputstream adaptive - it doesn't work
            if False and opt != 'pvr' and 'm3u8' in liz.getPath().lower() and inputstreamhelper.Helper('hls').check_inputstream():
                liz.setProperty('inputstream','inputstream.adaptive')
                liz.setProperty('inputstream.adaptive.manifest_type','hls')
                # liz.setProperty('inputstream.adaptive.media_renewal_url', 'plugin://%s/play/%s/%s'%(ADDON_ID,opt,id))
                # liz.setProperty('inputstream.adaptive.media_renewal_time', '900')
                #todo debug pvr (IPTV Simple) not playing with inputstream! temp. use kodiprops in m3u?
        xbmcplugin.setResolvedUrl(ROUTER.handle, found, liz)


Your test is interesting though.  Maybe sometimes the ad is needed to get the audio started. 

I will test some more.

**Comment by rocky4546** [comment]
Added the "False and" and retried Kodi.  Same result.  The video plays without sound unless the locast ad plays first.

**Comment by cookieisland** [comment]
I did some further testing. The DABL channel is working today but a few others have problems with no audio. I found that when I tried those channels using the Lunatix addon a couple worked but most had some problems with audio dropouts, video dropouts and also experienced the situation when a channel worked after an ad but failed to work if there was no add.

To me, this points to the Locast stream having issues as you noted previously.....

Hopefully Locast cleans up these stream problems in the future.
